pkg/addonfactory/template_agentaddon.go

```diff
@@ -46,10 +48,27 @@ func (a *TemplateAgentAddon) Manifests(
 	if err != nil {
 		return objects, err
 	}
-	for file, data := range a.templateData {
-		raw := assets.MustCreateAssetFromTemplate(file, data, configValues).Data
+
+	// sort the filenames of the templates so the manifests are ordered consistently
+	keys := make([]string, 0, len(a.templateData))
```

Comment:
why we need templateData to be a map initially?

Reply:
I get the data of template file from FS and template dir when build the agent, save the `file name : data` into the map.
the file name is used in `MustCreateAssetFromTemplate`.
otherwise, need save file name list and FS into the agentAddon struct, and get the data of template in `agentAddon.Manifests`. 
let me sort the file name when build the agent.

Reply:
changed to save templateFiles in the struct, and use the templatefiles order to Iterate the templateData.

Reply:
how about having a

```go
type templateFile struct {
  name string
  content []byte
}
```

Reply:
this definition is better. done.